src/overlays/actors/ovl_En_Dodongo/z_en_dodongo.c

```diff
@@ -136,97 +280,839 @@ static DamageTable D_808792D4 = {
+
+    if (yDiff >= 0) {
+        sp18 = &D_060042C4;
+        this->unk_306 = -this->unk_306;
+    } else {
+        sp18 = &D_06003B14;
+    }
+
+    Audio_PlayActorSound2(&this->actor, NA_SE_EN_DODO_J_TAIL);
+    SkelAnime_ChangeAnimPlaybackStop(&this->skelAnime, sp18, 2.0f);
+    this->unk_302 = 0;
+    this->collider1.base.atFlags |= AT_ON;
+    this->unk_304 = -1;
+    this->actionFunc = func_80878424;
+}
 
-extern ColliderJntSphElementInit D_808790D0[10];
-extern ColliderJntSphInit D_80879238;
-extern ColliderJntSphInit D_80879248;
-extern ColliderJntSphElementInit D_80879258[3];
-extern ColliderJntSphInit D_808792C4;
-extern DamageTable D_808792D4;
-extern CollisionCheckInfoInit D_808792F4;
-extern InitChainEntry D_808792FC[];
+void func_80878424(EnDodongo* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+    Vec3f sp20;
 
-extern UNK_TYPE D_060013C4;
-extern UNK_TYPE D_06001A44;
-extern UNK_TYPE D_060028F0;
-extern UNK_TYPE D_06003088;
-extern UNK_TYPE D_060042C4;
-extern UNK_TYPE D_06004C20;
-extern UNK_TYPE D_06008B1C;
+    this->unk_302++;
+    if (SkelAnime_FrameUpdateMatrix(&this->skelAnime)) {
+        if (!(player->stateFlags1 & 0x800000) && (Player_GetMask(globalCtx) != PLAYER_MASK_STONE_MASK)) {
+            this->collider1.base.atFlags &= ~AT_ON;
+            func_808777A8(this);
+        } else {
+            func_80877494(this);
+        }
+        return;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Dodongo/EnDodongo_Init.s")
+    this->actor.world.rot.y += this->unk_306;
+    this->actor.shape.rot.y = this->actor.world.rot.y;
+    sp20.x = this->collider1Elements[1].dim.worldSphere.center.x;
+    sp20.y = this->collider1Elements[1].dim.worldSphere.center.y;
+    sp20.z = this->collider1Elements[1].dim.worldSphere.center.z;
+    func_80876930(this, globalCtx, &sp20);
+    sp20.x = this->collider1Elements[2].dim.worldSphere.center.x;
+    sp20.y = this->collider1Elements[2].dim.worldSphere.center.y;
+    sp20.z = this->collider1Elements[2].dim.worldSphere.center.z;
+    func_80876930(this, globalCtx, &sp20);
+    CollisionCheck_SetAT(globalCtx, &globalCtx->colChkCtx, &this->collider1.base);
+    this->actor.flags |= 0x1000000;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Dodongo/EnDodongo_Destroy.s")
+void func_80878594(EnDodongo* this) {
+    this->actionFunc = func_808785B0;
+    this->actor.speedXZ = 0.0f;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Dodongo/func_80876930.s")
+void func_808785B0(EnDodongo* this, GlobalContext* globalCtx) {
+    if (this->unk_302 != 0) {
+        this->unk_302--;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Dodongo/func_80876B08.s")
+    if (this->unk_302 == 0) {
+        func_80876D28(this, globalCtx);
+        if (this->actor.colChkInfo.health == 0) {
+            func_80878724(this);
+        } else if (this->actor.xzDistToPlayer > SQ(10.0f) * this->unk_334) {
```

Comment:
```suggestion
        } else if (this->actor.xzDistToPlayer > 100.0f * this->unk_334) {
```
I don't think a square makes sense here since it's checking the absolute value and not its square.